lib/mongo/server/connection_pool/connection_pool_populator.rb

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2014-2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Server
+    # A manager that maintains the invariant that the
+    # size of a connection pool is at least minPoolSize.
+    #
+    # @api private
+    class ConnectionPoolPopulator
+      def initialize(pool)
+        @pool = pool
+        @thread = nil
+      end
+
+
+      def start!
+        @thread = Thread.new {
+          while !@pool.closed? do
+            @pool.populate
+            @pool.populate_semaphore.wait(nil)
+          end
+        }
+      end
+
+      def stop!
```

Comment:
The stop method should take a `wait=false` argument like the server monitor does. https://github.com/mongodb/mongo-ruby-driver/pull/1415 expands the scope of what is waited for,  I suggest waiting until that PR is merged prior to making this change.

Reply:
Which also means that, perhaps, `Pool#close` method should take a `wait: false` argument as well.

When clients are closed in test suite, the caller wants to wait for everything that is part of the client to be torn down and stopped which includes all background processing.

Reply:
Yes, I agree with this, since when pools are closed in the test suite we will want stop the background thread, etc. However, this won't help us when a client is closed, since the `Pool#close` method is not called (at least for now).